packages/frontend-shared/src/utils/getUrlWithParams.ts

```diff
@@ -5,6 +5,15 @@ export type LinkWithParams = {
 
 export const getUrlWithParams = (link: LinkWithParams) => {
   let result = link.url
+  const hasUtmParams = Object.keys(link.params).find((param) => param.startsWith('utm_'))
```

Comment:
Can use `some` to be more clear about the intention. When using `find` I usually expect the found variable is used, or an error is throw if it's not found. This is more "does it exist" in a true/false sense.

```suggestion
  const hasUtmParams = Object.keys(link.params).some((param) => param.startsWith('utm_'))
```

This is how Lodash's `findKey` works. https://github.com/lodash/lodash/blob/master/findKey.js#L23-L36

Or, if you want to be optimal and save a loop...

```js
function keyStartsWith (o, key) {
  for (const k in o) {
    if (k.startsWith(key)) return true
  }
  return false
}
```

I think `some` is the best candidate among `findKey`, `find` or the for loop.

Reply:
oh that's awesome, I always forget about `some` but it's ideal here